packages/edit-site/src/components/global-styles/screen-revisions/index.js

```diff
@@ -55,13 +55,37 @@ function ScreenRevisions() {
 	const { setEditorCanvasContainerView } = unlock(
 		useDispatch( editSiteStore )
 	);
+	const selectedRevisionMatchesEditorStyles = areGlobalStyleConfigsEqual(
+		globalStylesRevision,
+		userConfig
+	);
+	const shouldSelectFirstItem =
+		! selectedRevisionMatchesEditorStyles &&
+		( ! selectedRevisionId || 'unsaved' === selectedRevisionId );
 
 	useEffect( () => {
 		if ( editorCanvasContainerView !== 'global-styles-revisions' ) {
 			goBack();
 			setEditorCanvasContainerView( editorCanvasContainerView );
+			return;
+		}
+
+		/*
+			Ensure that the first item is selected where no revision is selected
+			and the selected styles don't match the current editor styles.
+			This is required in case editor styles are changed outside the revisions panel,
+			e.g., via the reset styles function of useGlobalStylesReset().
+			See: https://github.com/WordPress/gutenberg/issues/55866
+		*/
+		if ( shouldSelectFirstItem ) {
+			setGlobalStylesRevision( revisions[ 0 ] );
```

Comment:
Does this also need to call `setSelectedRevisionId` as in `selectRevision` or is it enough to just update the revision?

Reply:
It's enough to update the revision as all we want to do is load the preview in the frame.

Having said that, I don't think it'd hurt to just call `selectRevision()` for consistency's sake.

👍🏻